### DISCUSSION TITLE
Dockerfile seems to not copy docker-entrypoint.sh

### DISCUSSION THREAD

**Discussion by eebydeebygirl** [LBPUnion/ProjectLighthouse#993] [Q&A] [answered]
The game-server, API, and website containers don't seem to have access to the docker-entrypoint.sh file that gets copied from the build section of the dockerfile. I'm unsure if this is an error on my part or is a bug in the build process. 
The containers continuously reboot and log the error  `exec /lighthouse/docker-entrypoint.sh No such file or path `

If I could get pushed in the correct direction, I'd greatly appreciate it. Thank you!

**Comment by Slendy** [comment]
I am unable to replicate this issue on the latest commit. What docker version are you running and what plugins do you have installed?

**Reply by eebydeebygirl** [reply]
Just cloned the main the branch and attempted two different builds from the root using `docker build -t lighthouse .` & `docker compose up -d` and `docker build .` & `docker compose up -d` ; still got the same error.

My Docker version is 25.0.3, Docker Compose is 2.24.6-desktop.1. At the moment I have no additional plugins installed.

**Reply by Slendy** [reply]
Do you happen to be using Docker on Windows? I've never personally tested Lighthouse on anything other than Linux or MacOS but I'm not sure why the build would succeed and not copy the entrypoint script. What does the `/lighthouse` folder look like in the image if you create a container with `docker run --rm -it --entrypoint /bin/ash lighthouse`?

**Reply by eebydeebygirl** [reply]
I am using Docker on Windows. Running this, the folder `lighthouse` exists, and `docker-entrypoint.sh` exists.
The contents of the file look good too.

**Reply by Slendy** [reply, marked as answer]
Okay, so I've narrowed down the issue to the entrypoint script having CRLF line endings instead of LF line endings. I've added a .gitattributes file to the repository which should indicate to Git that the scripts need to maintain LF line endings when being cloned. Try re-cloning the repository now and running docker compose.

**Reply by eebydeebygirl** [reply]
I was just in the middle of setting up nested virtualization to test out the process on linux 😂 

Sweet, error has changed to something a bit more helpful haha
So right now it's saying the db is unavailable. I've edited the config file to change the db connection string to avail.

**Reply by Slendy** [reply]
The docker documentation is pretty bad right now, but on my local docker setup my lighthouse config looks like this:
```yaml
websiteListenUrl: http://website:10060
gameApiListenUrl: http://gameserver:10061
apiListenUrl: http://api:10062
dbConnectionString: server=db;uid=root;pwd=lighthouse;database=lighthouse
redisConnectionString: redis://redis:6379
```
Rather than using the IP addresses of the containers, the default docker network has DNS entries for each service.

**Reply by eebydeebygirl** [reply]
Aha! That did it. Thanks a bunch. I really appreciate it.

**Comment by sudokoko** [comment]
You should not need to build the image manually before doing a docker
compose up. It should just build for you

**Reply by eebydeebygirl** [reply]
Same error, unfortunately.
